solidity/contracts/deposit/DepositUtils.sol

```diff
@@ -489,8 +489,10 @@ library DepositUtils {
         bool frtExists = feeRebateTokenHolder(_d) != address(0);
 
         uint256 frtAdjustment = computeFrtAdjustment(_d, redeemerOwesNoFee, redeemerHoldsFrt, frtExists);
-        uint256 signerFeeAdjustment = computeSignerFeeAdjustment(_d, redeemerOwesNoFee, frtExists);
-
+        uint256 signerFeeAdjustment = computeSignerFeeAdjustment(_d, frtExists);
+        if(depositOwner(_d) != _redeemer){
+            return 0;
+        }
```

Comment:
Not following this bit...

Reply:
understandable, I also wanted a sanity check here.
If we don't return 0 here and redeemer is deposit owner, we could have a situation where `getRedemptionTbtcRequirement` returns a value greater than lotSize (eg: when the fee rebate token does not exist and TDT owner is the redeemer)

to be clear, I really don't like it here

Reply:
I think the only scenario here is if a non-TDT holder (owes lot size) redeems pre term/non courtesy call (owes fee) and there is no escrow (no FRT adjustment). But we prohibit that scenario because only the TDT holder is allowed to redeem pre-term. Does that sound right? See the table I linked above for the scenarios, as well.

Reply:
This is regarding at-term and courtesy-call cases.
If redeemer is not TDT owner, and FRT does not exist.
`computeSignerFeeAdjustment` just checks FRT existance and returns fee is it doesn't exist.
so `computeBaseRedemptionCharge` will return `LostSize`, and without the check `computeRedemptionFeeAdjustments` will return `signerFee`. the fee shouldn't be greater then `lotSize`

Reply:
I see. Let me think through this a little more.

Also, I just realized `computeFrtAdjustment` and `computeSignerFeeAdjustment` have their names reversed, , which is making this harder to reason about. Let's fix that.

Reply:
Reconsidering the whole thing here, how about we drop this FRT adjustment vs signer fee adjustment thing altogether, and instead go for a single `computeSignerFee` function:

```solidity
function computeSignerFee(
    DepositUtils.Deposit storage _d,
    bool _thirdPartyRedeemerOwesNoFee,
    bool _redeemerHoldsFrt,
    bool _frtExists
) internal view returns (uint256) {
    bool signerFeeNotOwed;
    if (_preTerm) {
        signerFeeNotOwed = _redeemerHoldsFrt;
    } else {
        signerFeeNotOwed = _frtExists || _redeemerIsThirdParty;
    }

    if (signerFeeNotOwed) {
        return 0;
    } else {
        return signerFee(_d);
    }
}
```

Reply:
This sounds like a great change overall